MoreLinq.Test/ReadOnlyDecoratedList.cs

```diff
@@ -0,0 +1,25 @@
+namespace MoreLinq.Test
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// A decorator for a <see cref="IReadOnlyList{T}"/> that ensures it does not implement another interface.
+    /// </summary>
+    sealed class ReadOnlyDecoratedList<T> : IReadOnlyList<T>
```

Comment:
If you're being strict then why not just have it implement `IReadOnlyCollection<>`?

Reply:
I can update the text. The meaning is that it doesn't implement `ICollection<>` or `IList<>`. so that any tests will use the `IReadOnlyCollection<>` or `IReadOnlyList<>` optimization.

Reply:
>  or `IReadOnlyList<>` optimization

Let's cross that bridge when we get there but not prepare in anticipation of a future that may never occur. Strictness is better here.